### ISSUE TITLE
Phalcon 2.0.* - creating object does not set model snapshots.

### ISSUE THREAD

**Issue by temuri416** [phalcon/cphalcon#11007]
Hi,

My model:

``` php
class Robots extends Model
{
    public function initialize()
    {
        $this->keepSnapshots(true);
    }
}
```

Immediately after creating a new instance of model, its snapshot is blank.

That prevents me from checking whether a field has been updated if I decide to update it later during the same request.

Snapshots only get set when loading a model. But they should also be set after a new model is created.

Thanks!

**Comment by ashpumpkin** [comment]
+1, but I think it's not a bug, but a very annoying feature. Would be very cool if someone will fix it

**Comment by temuri416** [comment]
Snapshot should also be refreshed when Model is saved/updated.

**Comment by web777** [comment]
Old issue. From 1.3.x

**Comment by ruudboon** [comment]
Refreshing the object doesn't help.

```
<?php
$model = new MyModel();
$model->name = "test";
$model->save();
$model->refresh();
```
I solved this by completely reloading the object :(
````
<?php
$model = new MyModel();
$model->name = "test";
$model->save();
$model = MyModel::findFirst($model->payment_id);
```

**Comment by sergeyklay** [comment]
@ruudboon yeah :confused: https://github.com/sergeyklay/incubator/blob/2.0.x/Library/Phalcon/Mvc/Model/Behavior/NestedSet.php#L338

**Comment by ruudboon** [comment]
I'm sorry @sergeyklay I don't understand your feedback.

**Comment by sergeyklay** [comment]
I mean that this is a known issue and I too just had to use this trick

**Comment by ruudboon** [comment]
Ah sorry! Now I see. Completely clear. Hope to get some time soon to contribute to Phalcon. Love this project.

**Comment by temuri416** [comment]
@andresgutierrez 

Any update on this? Do you plan to implement it in 2.1?

Thanks!

**Comment by andrew-demb** [comment]
Up. 
This will be very helpful for development, but if now implement the feature it can break some applications.

@andresgutierrez, can you give us a feedback about this?

**Comment by temuri416** [comment]
@sergeyklay Is this on your radar? :)

**Comment by Jurigag** [comment]
No because 2.0.x is no longer supported.

**Comment by temuri416** [comment]
Sorry, I wasn't expecting this to be fixed in 2.0.x :-) Any hope for this go to on 3.x branch?

**Comment by Mechzeit** [comment]
My work around for now is using a base mode class, with the 'afterCreate' method
Obviously it's not perfect, it doesn't work with protected properties - but it's my solution for now.

```php
<?php
class SomeModel extends BaseModel {
    //...

    public function afterCreate()
    {
        // I'm using parent because the child classes overload toArray(). For most, $this->toArray() would suffice
        $this->setSnapshotData(parent::toArray());
    }
}
```

**Comment by Jurigag** [comment]
I made PR with fix for this, let me know what you think.

**Comment by sergeyklay** [comment]
Fixed in the `3.1.x` branch.
